2015/10-dancer-i18n-with-log-report.pod

```diff
@@ -0,0 +1,206 @@
+=encoding utf8
+
+=head1 Implementing i18n in a Dancer application using Plugin::LogReport
+
+=head2 Introduction
+
+There are a number of ways to approach the internationalisation of a web
+application. I'm going to look at a method using L<Log::Report>, which includes
+very comprehensive translation functionality.
+
+More specifically, I will use its Dancer2 plugin,
+L<Dancer2::Plugin::LogReport>. If you haven't already, please look at my other
+article on that.
+
+=head2 A basic application
+
+Let's get started by creating a basic application with some strings that we want
+translated. We'll look at translating the content of templates later.
+
+ # In your route handler
+ package MyApp;
+ use Dancer2;
+ use Dancer2::Plugin::LogReport 'my-domain'; # domain used to group sets of messages
+ 
+ get '/' => sub {
+   notice "Hello world";
+   template 'index';
+ };
+
+ # In the index template
+ [% FOREACH message IN session.messages %]
+   <p>[% message.toString %]</p>
+ [% END %]
+
+By running this application, you should see C<Hello world> on the index page.
+
+=head2 Translating strings
+
+L<Log::Report> has L<a lot of functions to translate
+strings|Log::Report#Language-Translations>. We'll touch on 2 of the commonly
+used ones here.
+
+Firstly to simply translate a fixed string into another, we can use C<__> (2
+underscores):
+
+ notice __"Hello world";
+
+This might look a bit strange, but the 2 underscores are simply a function. You
+can wrap the text in parentheses if you prefer:
+
+ notice __("Hello world");
+
+They work like any other function, although be aware that you can't use single
+quotes, as C<__'> is interpreted as C<::> by Perl
+
+We can also use C<__x>, which is the same as C<__>, except that it allows for
+variables within the string. For example:
+
+ notice __x"Hello {name}", name => 'Andy';
+
+This means that when the text is extracted as message IDs, that any variable
+content is conveniently separated. It also allows all sorts of clever things,
+such as adjusting the text for pluralisation depending on variable conditions,
+but I won't go into that just yet.
+
+So, for now, update your web application accordingly:
+
+ get '/' => sub {
+   notice __"Hello world";
+   notice __x"Hello {name}", name => 'Andy';
+   template 'index';
+ };
```

Comment:
I'd actually prefer to leave it as it is if possible, because it's the standard way that both Log::Report and Locale::TextDomain encourage. It makes the code more readable, especially when there is a large quantity of text to be translated (hence the reason why such a short unobtrusive function name was chosen by the modules).